streams/src/main/java/org/apache/kafka/streams/processor/internals/StreamTask.java

```diff
@@ -482,7 +482,7 @@ public void closeSuspended(boolean clean,
             if (eosEnabled) {
                 if (!clean) {
                     try {
-                        if (!isZombie) {
+                        if (!isZombie && transactionInFlight) {
```

Comment:
This comment is not for this line but for line 361 above (it is not introduced in this PR but we can tighten the screws a bit more along side):

we should switch the line `transactionInFlight = true;` to go after `producer.beginTransaction();`